lib/Models/CatalogNew.ts

```diff
@@ -1,12 +1,67 @@
 import Terria from "./Terria";
 import CatalogGroup from "./CatalogGroupNew";
+import { computed } from "mobx";
+import { USER_ADDED_CATEGORY_NAME } from "../Core/addedByUser";
+import isDefined from "../Core/isDefined";
+import hasTraits from "./hasTraits";
+import CatalogMemberTraits from "../Traits/CatalogMemberTraits";
+import Model, { BaseModel } from "./Model";
+import CommonStrata from "./CommonStrata";
+import ModelReference from "../Traits/ModelReference";
+
+class UserAddedDataGroup extends CatalogGroup {
```

Comment:
Do we need a type for this? I think these two methods could be added to `GroupMixin`. Should change them to take a `stratumId` as a parameter too, rather than assuming `CommonStrata.user`. And mark them `@action`. And it should be ok to `push` directly into the `members` array (creating it first if necessary) rather than using concat.